### PR TITLE
Templates service refactor

### LINKED ISSUE #924: Move template storage from AWS to db & s3
**Is your feature request related to a problem? Please describe.**
Currently, the templates that are used to send e-mails, are being stored in AWS as part of their SES [templates feature](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-personalized-email-api.html), and we use the AWS api to create, update, delete and list templates.

The code for the CRUD operations reside in the `storage/templates` package, we'll want to move that to the upper layer. Our API endpoints for the `templates` resource should remain roughly the same, except maybe the `List` operation, we should adapt it to be the same as the rest of the paginated resources.

The templates that are created, are being used as part of the campaign operation. When we create a campaign we associate a template, to be rendered as an e-mail message when we send it to each subscriber. Templates are used as a way to send personalized e-mail to the customers, the subject and body can contain unique values that are personalized for each recipient.

**Describe the solution you'd like**

Let's start with the storage. A template consists of:
- A name
- Subject
- Text
- HTML

Text and html parts are the actual body, we can have an html template or send just a plain text message. Here are some example values:

```json
{
    "name": "MyTemplate",
    "subject": "Greetings, {{.name}}!",
    "html": "<h1>Hello {{.name}},</h1><p>Your favorite animal is {{.favoriteanimal}}.</p>",
    "text": "Dear {{.name}},\r\nYour favorite animal is {{.favoriteanimal}}."
}
```

The html and text part must be a valid golang template i.e. parsable by the [`text/template`](https://pkg.go.dev/text/template) package.

The table should look something like this:

|   Name    |            Type             | Nullable |
|-----------|-----------------------------|----------|
| id        | bigint unsigned primary key | false    |
| name      | varchar(191)                | false    |
| subject   | varchar(191)                | false    |
| text_part | text                        | true     |

The HTML part will be stored on s3 instead of the database since it's more suiting that way. We'll figure out the name of the bucket and folder later on. In case the user wants to send only a text email, not a _fancy_ HTML one, we can skip that part.

Next up is the storage interface and implementation. Currently, this resides in the `storage/templates`, we should refactor this into our existing Storage interface, and re-implement all CRUD operations with the new store methods. We should also include tests since we don't have them because currently, it is a simple AWS API wrapper.

The final step, and probably the hardest one, is to refactor the send operation to use the templates from our end instead of the `SendTemplatedEmail` and `SendBulkTemplatedEmail` APIs.

### PR DESCRIPTION
**Pull request by Dzalevski**
# Pull Request Template

## Description

Rework CRUD Templates to work with mysql/s3

Fixes #924

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Dzalevski** [comment]
@gudgl we need to  change subject -> subject_part

File: storage/s3/storage.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+)
+
+const key = "s3"
+
+type S3Storage interface {
```
Comment (FilipNikolovski) [inline storage/s3/storage.go:11]: We can use s3iface.S3API instead of this right?

Comment (gudgl) [reply storage/s3/storage.go:11]: @FilipNikolovski can you please explain whats the difference

File: storage/migrations/sqlite3/4_templates.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+-- +migrate Up
+
+CREATE TABLE IF NOT EXISTS "templates"
+(
+    "id"         integer primary key autoincrement,
+    `user_id`    integer unsigned NOT NULL,
+    `name`       varchar(191)     NOT NULL,
+    "subject"    varchar(191)     NOT NULL,
```
Comment (FilipNikolovski) [inline storage/migrations/sqlite3/4_templates.sql:8]:
```suggestion
    "subject_part"    varchar(191)     NOT NULL,
```

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**